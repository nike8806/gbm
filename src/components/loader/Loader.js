import React from 'react';
import './Loader.scss';

/**
 * Loader component
*/
const Loader = () => (
  <div className="spinner">
    <div className="spinner__mask" />
    <div className="spinner__logo" />
  </div>
);

export default Loader;
