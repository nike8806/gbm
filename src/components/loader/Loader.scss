.spinner {
    &__mask {
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      background-color: #ffffff;
      opacity: 0.50;
    }
  
    &__logo {
      background-image: url(./Loader.gif);
      background-repeat: no-repeat;
      background-position: center center;
      position: absolute;
      margin: -74px -74px;
      width: 148px;
      height: 148px;
      top: 50%;
      left: 50%;
    }
  }
  