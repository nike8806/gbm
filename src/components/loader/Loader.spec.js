import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Loader from './Loader';

let component;

describe('Loader component', () => {
  describe('with no theme', () => {
    beforeEach(() => {
      component = shallow(
        <Loader />
      );
    });

    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
