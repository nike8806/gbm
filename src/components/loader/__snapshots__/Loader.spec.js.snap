// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component with no theme should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="spinner__mask"
        />,
        <div
          className="spinner__logo"
        />,
      ],
      "className": "spinner",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "spinner__mask",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "spinner__logo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="spinner__mask"
          />,
          <div
            className="spinner__logo"
          />,
        ],
        "className": "spinner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spinner__mask",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spinner__logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
